### PR TITLE
Implement context-aware RunEventLoop

### PR DESCRIPTION
**Pull request by lololozhkin**
implemented RunEventLoopContext method. now socketmodehandler will react on context cancellation

##### PR preparation

- [x] Run `make pr-prep` from the root of the repository to run formatting, linting and tests.

### CODE REVIEW

**Comment by lololozhkin** [comment]
resolves slack-go/slack#1162

**Comment by lololozhkin** [comment]
Hi, added context to run method on socketmodehandler. make pr-prep was successful. Could you please merge it in master, to resolve slack-go/slack#1162 ?

**Comment by lololozhkin** [comment]
and, could you please approve workflow run? It's my first PR..

File: socketmode/socketmode_handler.go [hunk 1]
```diff
@@ -105,15 +106,31 @@ func (r *SocketmodeHandler) HandleDefault(f SocketmodeHandlerFunc) {
 // RunSlackEventLoop receives the event via the socket
 func (r *SocketmodeHandler) RunEventLoop() error {
 
-	go r.runEventLoop()
+	go r.runEventLoop(context.Background())
 
 	return r.Client.Run()
 }
 
+func (r *SocketmodeHandler) RunEventLoopContext(ctx context.Context) error {
```
Comment (DestroyerAlpha) [inline socketmode/socketmode_handler.go:114]:
```suggestion
func (r *SocketmodeHandler) RunEventLoopWithContext(ctx context.Context) error {
```

**Comment by sbansal7** [comment]
Hi @lololozhkin any chance you could respond to the above comment? -Thanks!

**Review by kanata2 — APPROVED** [review]

**Comment by kanata2** [comment]
Thanks! LGTM 👍

File: socketmode/socketmode_handler.go (hunk 1 above)
Comment (kanata2) [reply socketmode/socketmode_handler.go:114]: Since naming convention for context-aware methods has not been determined, leave it as it is in this time.

**Pull request merged**